grails-app/controllers/au/org/ala/userdetails/RegistrationController.groovy

```diff
@@ -150,8 +150,20 @@ class RegistrationController {
 
         if (user) {
             if (params.email != user.email) {
-                // email address has changed, and username and email address must be kept in sync
+                // email address has changed
+                if (userService.isEmailInUse(params.email, user)) {
+                    render(view: "accountError", model: [msg: "Failed to update user profile - A user is already registered with the email address"])
+                    return
+                }
+                // and username and email address must be kept in sync
                 params.userName = params.email
+                // update cookie when email address changed https://github.com/AtlasOfLivingAustralia/userdetails/issues/98
+                // cookie config values need to be consistent with values in CAS
+                def cookieName = grailsApplication.config.ala.cookie.name
+                if (cookieName && cookieService.findCookie(cookieName)) {
+                    cookieService.setCookie(cookieName, quoteValue(params.email), grailsApplication.config.ala.cookie.maxAge, grailsApplication.config.ala.cookie.path,
```

Comment:
Minor nitpick but I dislike the grailsApplication.config.foo.bar style, I prefer either injecting with `@Value` or using `grailsApplication.config.getProperty('foo.bar', Type, optionalDefault)`

Reply:
updated